src/SixLabors.Fonts/Tables/Cff/CffEvaluationEngine.cs

```diff
@@ -0,0 +1,747 @@
+// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using System.Collections.Generic;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+
+namespace SixLabors.Fonts.Tables.Cff
+{
+    /// <summary>
+    /// Decodes the commands and numbers making up a Type 2 CharString. A Type 2 CharString extends on the Type 1 CharString format.
+    /// Compared to the Type 1 format, the Type 2 encoding offers smaller size and an opportunity for better rendering quality and
+    /// performance. The Type 2 charstring operators are (with one exception) a superset of the Type 1 operators.
+    /// </summary>
+    /// <remarks>
+    /// A Type 2 charstring program is a sequence of unsigned 8-bit bytes that encode numbers and operators.
+    /// The byte value specifies a operator, a number, or subsequent bytes that are to be interpreted in a specific manner
+    /// </remarks>
+    internal ref struct CffEvaluationEngine
```

Comment:
@brianpopow this is a complete rewrite of the evaluator based upon the Fontkit implementation and sense checked against PDFPig. It's much, much more efficient that the original implementation saving a secondary buffer allocation for each set of operand instructions in the font.